docs/en/serving/api_server.md

```diff
@@ -173,6 +173,49 @@ for message in messages:
         print(item)
 ```
 
+### Tools
+
+Currently, LMDeploy supports tools only for InternLM2 models. Please start the service of InternLM2 models before running the following example.
+
+```python
+from openai import OpenAI
+
+tools = [
+  {
+    "type": "function",
+    "function": {
+      "name": "get_current_weather",
+      "description": "Get the current weather in a given location",
+      "parameters": {
+        "type": "object",
+        "properties": {
+          "location": {
+            "type": "string",
+            "description": "The city and state, e.g. San Francisco, CA",
+          },
+          "unit": {"type": "string", "enum": ["celsius", "fahrenheit"]},
+        },
+        "required": ["location"],
+      },
+    }
+  }
+]
+messages = [{"role": "user", "content": "What's the weather like in Boston today?"}]
+tool_choice={"type": "function", "function": {"name": "get_current_weather"}}
```

Comment:
Do we support `tool_choice`?

Reply:
Only supports `none` or specifying a particular tool in json format.